sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/DiagnosticAnalyzerContextHelper.cs

```diff
@@ -40,7 +40,7 @@ internal static class DiagnosticAnalyzerContextHelper
             context.RegisterSyntaxNodeAction(
                 c =>
                 {
-                    if (!c.GetSyntaxTree().IsGenerated(generatedCodeRecognizer, c.SemanticModel.Compilation))
+                    if (ShouldAnalyze(generatedCodeRecognizer, c.GetSyntaxTree(), c.Compilation, c.Options))

```

Comment:
Need at least one integration test (in C#) that check the end-to-end analysis i.e. that the analyzers actually analyze/skip based on the setting in SonarLint.xml.

Reply:
fair point, I only added one for the plugin